### PR TITLE
[DOCS] Remove some api 1.0 names and mentions from dev docs

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Remove some api 1.0 names and mentions from dev docs
 
### Tickets:
 - CVS-132237

### CODE REVIEW

File: src/core/docs/api_details.md [hunk 1]
```diff
@@ -7,11 +7,11 @@ OpenVINO Core API contains two folders:
 <pre>
  <code>
  <a href="../include/openvino">openvino/</a>                  // Common folder with OpenVINO 2.0 API
```
Comment (ilya-lavrenov) [inline src/core/docs/api_details.md:9]:
```suggestion
 <a href="../include/openvino">openvino/</a>                  // Common folder with OpenVINO API
```

Comment (olpipi) [reply src/core/docs/api_details.md:9]: So all mentions of "OpenVINO API 2.0" (and their variations) should be replaced by "OpenVINO API"?

Comment (ilya-lavrenov) [reply src/core/docs/api_details.md:9]: yes, because we now have only 1 API :)

**Review by sgolebiewski-intel — APPROVED** [review]

File: src/core/docs/tests.md [hunk 2]
```diff
@@ -3,10 +3,10 @@
 OpenVINO Core is covered by tests from the binary `ov_core_unit_tests`. This binary can be built by the target with the same name.
 
 OpenVINO Core [tests](../tests/) have next structure:
+ * `.` - common core tests + op evaluation tests
```
Comment (vurusovs) [inline src/core/docs/tests.md:6]:
```suggestion
OpenVINO Core [tests](../tests/) consists of common and op evaluation tests. Folder have next structure:
```

Comment (olpipi) [reply src/core/docs/tests.md]: вщту

File: src/core/docs/api_details.md (hunk 1 above)
Comment (olpipi) [reply src/core/docs/api_details.md]: done

File: src/tests/README.md [hunk 3]
```diff
@@ -9,42 +9,42 @@ This is OpenVINO testing framework. OpenVINO test system contains:
...
+    `IPlugin` class only.
 
   * It is not prohibited to have several test files for the same file from the tested module.
   * It is not prohibited to create a separate test file for specific classes or functions (not for the whole file).
 
 * **Functional tests**
   This test type is used to verify public OpenVINO API. There are following types of functional tests:
-  * `inference_engine_tests` are plugin-independent tests. They are used to verify OpenVINO API methods that do not
-    involve any plugin runtime. The examples are: `network_reader`, `network_serializer`, and `precision` tests.
+  * `openvino_tests` are plugin-independent tests. They are used to verify OpenVINO API methods that do not
```
Comment (vurusovs) [inline src/tests/README.md:34]: Originally, `inference_engine_tests` and `plugin_tests` were mean folders with specific tests. For the moment we have `functional/plugin/`, `functional/.../single_op/` and `functional/.../subgraph/`. In this case we need to actualize the document, but may be done in a separate PR

**Review by ilya-lavrenov — APPROVED** [review]
